# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../xf86-video-vesa/xf86GTFMode.patch
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- xf86-video-vesa-2.0.0/src/vesa.c.orig	2008-10-22 09:40:13.000000000 +0200
+++ xf86-video-vesa-2.0.0/src/vesa.c	2008-10-22 09:41:15.000000000 +0200
@@ -342,7 +342,12 @@
      * then, by this point, you're well into despair territory.
      */
     for (v = mon->vrefresh[0].lo; v <= mon->vrefresh[0].hi; v++) {
+#if XORG_VERSION_CURRENT > XORG_VERSION_NUMERIC(1,4,2,0,0)
+	/* xf86GTFMode is only avalaible in newer versions of Xorg */
 	mode = xf86GTFMode(p->HDisplay, p->VDisplay, v, 0, 0);
+#else
+	mode = xf86CVTMode(p->HDisplay, p->VDisplay, v, FALSE, FALSE);
+#endif
 	ret = xf86CheckModeForMonitor(mode, mon);
 	xfree(mode);
 	if (ret == MODE_OK)
